litestar/dto/factory/_backends/types.py

```diff
@@ -10,38 +10,83 @@
-    """Name of the field as it should feature on the transfer model."""
+
+@dataclass
+class TransferModel:
+    """Type for representing fields and model type of nested model type."""
+
+    model: type[Any]
+    field_definitions: FieldDefinitionsType
 
 
 @dataclass
-class NestedFieldDefinition:
-    """For representing nested model."""
+class TransferType:
+    """Type for representing model types for data transfer."""
+
+    __slots__ = ("parsed_type",)
 
-    field_definition: TransferFieldDefinition
-    nested_type: Any
-    nested_field_definitions: FieldDefinitionsType = field(default_factory=dict)
+    parsed_type: ParsedType
 
-    @property
-    def name(self) -> str:
-        """Name of the field."""
-        return self.field_definition.name
 
-    @property
-    def serialization_name(self) -> str | None:
-        """Serialization name of the field."""
-        return self.field_definition.serialization_name
+@dataclass
+class SimpleType(TransferType):
+    """Represents indivisible, non-composite types."""
+
+    __slots__ = ("transfer_model",)
+
+    transfer_model: TransferModel | None
+    """If the type is a 'nested' type, this is the model generated for transfer to/from it."""
+
+
+@dataclass
+class CompositeType(TransferType):
+    """A type that is made up of other types."""
+
+    __slots__ = ("has_nested",)
+
+    has_nested: bool
+    """Whether the type represents nested model types within itself."""
+
+
+@dataclass
+class UnionType(CompositeType):
+    """Type for representing union types for data transfer."""
 
-    def make_field_type(self, inner_type: type) -> Any:
-        if self.field_definition.parsed_type.is_collection:
-            return self.field_definition.parsed_type.safe_generic_origin[inner_type]
-        if self.field_definition.parsed_type.is_optional:
-            return self.field_definition.parsed_type.safe_generic_origin[inner_type, None]
-        return inner_type
+    inner_types: tuple[TransferType, ...]
+
+
+@dataclass
+class CollectionType(CompositeType):
+    """Type for representing collection types for data transfer."""
+
+    inner_type: TransferType
+
+
+@dataclass
+class TupleType(CompositeType):
+    """Type for representing tuples for data transfer."""
+
+    inner_types: tuple[TransferType, ...]
+
+
+@dataclass
+class MappingType(CompositeType):
+    """Type for representing mappings for data transfer."""
+
+    key_type: TransferType
+    value_type: TransferType
+
+
+@dataclass(frozen=True)
+class TransferFieldDefinition(FieldDefinition):
+    transfer_type: TransferType
+    """Type of the field for transfer."""
+    serialization_name: str | None = field(default=None)
+    """Name of the field as it should feature on the transfer model."""
 
 
-FieldDefinitionsType: TypeAlias = "Mapping[str, TransferFieldDefinition | NestedFieldDefinition]"
+"""For typing where any field definition is allowed."""
+FieldDefinitionsType: TypeAlias = "Mapping[str, TransferFieldDefinition]"
```

Comment:
look into where we are actually utilizing the mapping aspect of this, off the top of my head we don't need it as the name of the field is on the field definition.

Reply:
was only using it in tests - changed it to `tuple[TransferFieldDefinition, ...]`